x-pack/metricbeat/module/statsd/server/data.go

```diff
@@ -102,26 +102,27 @@ func parse(b []byte) ([]statsdMetric, error) {
 	return metrics, nil
 }
 
-func eventMapping(metricName string, metricValue interface{}, metricSetFields mapstr.M, mappings map[string]StatsdMapping) {
+func eventMapping(metricName string, metricValue interface{}, mappings map[string]StatsdMapping) *mapstr.M {
```

Comment:
Any specific reason for using a pointer to a map? I see that you're trying to return a pointer to mapstr.M, which is a pointer to a pointer, which should never be required. Please consider not using a pointer type and making sure the tests still pass. 
Map types in Go are reference type just like pointers and slices. Unless there's a strong reason, you should avoid using pointers to Maps. [Ref](https://go.dev/blog/maps)

Reply:
Updated!